docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,200 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+Async programming is traditionally associated with callbacks and a general lack of straightforward code.  One might even say that straightforward code and scalable code are often at odds with one another.  This is not a problem with the C#/VB async model!  It is a language-level construct which hides complexity and allows you to focus on the problem at hand without making the code confusing in the process.
```

Comment:
It seems odd to refer it the model as the "C#/VB async model". Task is not a language thing. The syntax is.